com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UploadAppStep.java

```diff
@@ -57,10 +57,14 @@ public StepPhase executeAsyncStep(ExecutionWrapper execution) throws FileStorage
             String fileName = StepsUtil.getModuleFileName(execution.getContext(), app.getModuleName());
 
             getStepLogger().debug(Messages.UPLOADING_FILE_0_FOR_APP_1, fileName, app.getName());
+
+            if (fileName == null) {
+                return StepPhase.DONE;

```

Comment:
Add a debug message here, saying that there's no content to upload.

Reply:
Also, move this check before the debug message that's saying that we're uploading something.